### ISSUE TITLE
Improvement on example cd_chi2ks_adult

### ISSUE THREAD

**Issue by yanhong-zhao-ef** [SeldonIO/alibi-detect#223]
I was following the documentation with my own data here https://docs.seldon.io/projects/alibi-detect/en/latest/examples/cd_chi2ks_adult.html 

and noticed that the function here 
```
def plot_categories(idx: int) -> None:
    # reference data
    fref_drift = {cat: cd.x_ref_count[idx][i] for i, cat in enumerate(category_map[idx])}

    # test set
    cats = {f: list(np.unique(X_t1[:, f])) for f in categories_per_feature.keys()}
    X_count = {f: [(X_t1[:, f] == v).sum() for v in vals] for f, vals in cats.items()}
    fxt1_drift = {cat: X_count[idx][i] for i, cat in enumerate(category_map[idx])}

    # plot bar chart
    plot_labels = list(fxt1_drift.keys())
    ind = np.arange(len(plot_labels))
    width = .35
    fig, ax = plt.subplots()
    p1 = ax.bar(ind, list(fref_drift.values()), width)
    p2 = ax.bar(ind + width, list(fxt1_drift.values()), width)
    ax.set_title(f'Counts per category for {feature_names[idx]} feature')
    ax.set_xticks(ind + width / 2)
    ax.set_xticklabels(plot_labels)
    ax.legend((p1[0], p2[0]), ('Reference', 'Test'), loc='upper right', ncol=2)
    ax.set_ylabel('Counts')
    ax.set_xlabel('Categories')
    plt.xticks(list(np.arange(len(plot_labels))), plot_labels, rotation='vertical')
    plt.show()
```
Will throw an error when the test set is missing some categories in some columns compared to the reference data. By grabbing the __get_counts function from the TabularDrift fixes the problem
```
# test set
    X_count = {f: np.bincount(validation_data_raw.drop(columns=['Date', 'unique_id']).to_numpy()[:, f].astype(int), minlength=n_cat) for f, n_cat in cd.categories_per_feature.items()}
    fxt1_drift = {cat: X_count[idx][i] for i, cat in enumerate(category_map[idx])}
```
Should I open a PR to improve this function in the example? (sorry I have left my own data specific code in but you get the idea)

Thanks for the awesome package!

**Comment by arnaudvl** [comment]
Hi @yanhong-zhao-ef , that function was not intended to be a generic function but just to visualise some distributions in the existing example notebook. But any improvement is of course welcomed! The notebook will change in PR #222 but feel free to add a PR after #222 is merged and add in your changes.

**Comment by yanhong-zhao-ef** [comment]
awesome! just saw #222 and was going to suggest allowing unseen categories in test data in a separate issue yesterday!

thanks for the awesome work!
